xdsl/transforms/convert_ptr_to_riscv.py

```diff
@@ -91,23 +90,19 @@ def match_and_rewrite(self, op: ptr.LoadOp, rewriter: PatternRewriter, /):
 
         result_register_type = register_type_for_type(op.res.type)
 
-        match result_register_type:
-            case riscv.IntRegisterType:
-                lw_op = riscv.LwOp(addr, 0, comment="load word from pointer")
-            case riscv.FloatRegisterType:
-                float_type = cast(AnyFloat, op.res.type)
-                match float_type:
-                    case Float32Type():
-                        lw_op = riscv.FLwOp(addr, 0, comment="load float from pointer")
-                    case Float64Type():
-                        lw_op = riscv.FLdOp(addr, 0, comment="load double from pointer")
-                    case _:
-                        raise DiagnosticException(
-                            f"Unexpected floating point type {float_type}"
-                        )
-
-            case _:
-                assert False, f"Unexpected register type {result_register_type}"
+        if issubclass(result_register_type, riscv.IntRegisterType):
+            lw_op = riscv.LwOp(addr, 0, comment="load word from pointer")
+        else:
+            float_type = cast(AnyFloat, op.res.type)
+            match float_type:
+                case Float32Type():
+                    lw_op = riscv.FLwOp(addr, 0, comment="load float from pointer")
+                case Float64Type():
+                    lw_op = riscv.FLdOp(addr, 0, comment="load double from pointer")
+                case _:
+                    raise DiagnosticException(
+                        f"Unexpected floating point type {float_type}"
+                    )
```

Comment:
Can the register definitely not take a different type?

Reply:
Hmm, this should probably have a different message, as it's more of a not implemented situation, I'll add this.